src/main/java/org/qubership/itool/context/FlowContextImpl.java

```diff
@@ -74,27 +77,39 @@ public class FlowContextImpl implements FlowContext {
     private ClassLoader taskClassLoader;
     private Vertx vertx;
     private boolean breakRequested;
-
-    /* Create a flow context not bound to any graph service and create a new graph in it. */
-    public FlowContextImpl() {
-        this((GraphService)null);
+    private final GraphFactory graphFactory;
+    private final GraphReportFactory graphReportFactory;
+    private final GraphMergerFactory graphMergerFactory;
+
+    @Inject
+    public FlowContextImpl(GraphFactory graphFactory, GraphReportFactory graphReportFactory, GraphMergerFactory graphMergerFactory) {
+        this.graphFactory = graphFactory;
+        this.graphReportFactory = graphReportFactory;
+        this.graphMergerFactory = graphMergerFactory;
+        
+        this.report = graphReportFactory.createGraphReport();
+        this.graph = graphFactory.createGraph();
+        this.graph.setReport(report);
+        this.graphService = null;
     }
 
-    /* Create a flow context not bound to any graph service and use provided graph in it. */
-    public FlowContextImpl(Graph graph) {
+    public FlowContextImpl(GraphFactory graphFactory, GraphReportFactory graphReportFactory, GraphMergerFactory graphMergerFactory, Graph graph) {
+        this.graphFactory = graphFactory;
+        this.graphReportFactory = graphReportFactory;
+        this.graphMergerFactory = graphMergerFactory;
+        
         this.graph = graph;
         this.report = graph.getReport();
         this.graphService = null;
     }
 
-    /** Create a flow context and a new graph for it.
-     * If GraphService is provided, the given graph is registered in that service.
-     *
-     * @param graphService Graph Service, nullable
-     */
-    public FlowContextImpl(GraphService graphService) {
-        this.report = new GraphReportImpl();
-        this.graph = new GraphImpl();
+    public FlowContextImpl(GraphFactory graphFactory, GraphReportFactory graphReportFactory, GraphMergerFactory graphMergerFactory, GraphService graphService) {
```

Comment:
Are those extra constructors really needed? If they are needed, have you considered [constructor chaining](https://www.baeldung.com/java-chain-constructors) to reduce code duplication?